src/TLinkStore.cpp

```diff
@@ -50,6 +50,16 @@ QStringList& TLinkStore::getHints(int id)
     return mHintStore[id];
 }
 
+QStringList TLinkStore::getLinksConst(int id) const
+{
+    return mLinkStore[id];
```

Comment:
Given that you have declared this method and the next to be `const` you should not be using the non-constant `(QStringList) QMap<int, QStringList>::operator[]` - as if the particular `id` key is NOT present in the `QMap` ***then a default constructed (empty) `QStringList()` will be created and inserted INTO the `mLinkStore` under that key***, which will modify it and violate the `const`ness. As it happens there are both:
* `const (QStringList) QMap<int, QStringList>::operator[](const int& key) const`
* `(QStringList) QMap<int, QStringList>::operator[](const int& key)`

but you will be better off using the explicitly `const`:
`const (QStringList) QMap<int, QStringList>::value(const int& key, const QStringList& defaultValue = QStringList()) const`

I.e.:
```cpp
    return mLinkStore.value(id);
```